--- origsrc/pidgin-libnotify-0.13/src/Makefile.am	2007-06-30 13:52:52.000000000 -0500
+++ src/pidgin-libnotify-0.13/src/Makefile.am	2008-05-19 19:16:49.796875000 -0500
@@ -1,6 +1,6 @@
 gddir = $(LIBPURPLE_LIBDIR)/purple-2
 
-pidgin_libnotify_la_LDFLAGS = -module -avoid-version
+pidgin_libnotify_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if PLUGINS
 
@@ -10,7 +10,7 @@
 	pidgin-libnotify.c \
 	gln_intl.h
 
-pidgin_libnotify_la_LIBADD = $(LIBNOTIFY_LIBS) $(DBUS_LIBS) $(GTK_LIBS)
+pidgin_libnotify_la_LIBADD = $(LIBNOTIFY_LIBS) $(DBUS_LIBS) $(PIDGIN_LIBS)
 
 endif
 
